core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -318,6 +358,25 @@ class Partition(val topicPartition: TopicPartition,
 
   def inSyncReplicaIds: Set[Int] = partitionState.isr
 
+  def maybeAddListener(listener: PartitionListener): Boolean = {
+    inReadLock(leaderIsrUpdateLock) {
+      // `log` is set to `None` when the partition is failed or deleted.
+      log match {
+        case Some(log) =>
+          listener.onHighWatermarkUpdated(topicPartition, log.highWatermark)
+          listeners.add(listener)
```

Comment:
Sorry if I missed it, but why do we call onHighWatermarkUpdated here?

Reply:
I think that you are right. As a second thought, I don't need this call during the registration for the new group coordinator because it registers before doing any writes. I will remove it.

@jolshan The idea was to provide an initial value to the listener.